Automattic/jetpack
# Use the Identity_Crisis package

Update Jetpack to use the Identity_Crisis package. Included is migration of Sync Functions home_url, site_url and dependent functions to the Connections package so that we have the following package dependencies

Sync -> IDC + Connections
IDC -> Connections

Regression Testing of IDC is needed to  merge this PR.

#### Changes proposed in this Pull Request:

* Replaces the usage of Jetpack_IDC with Identity_Crisis.
* Maintains Jetpack_IDC class for backward compatibility during upgrades.
* Migrate Sync/Function url functions into Connection/Utils
* Remove Identity Crisis dependencies on Sync.
* Update Connection to use normalized site_url and home_url during registration flow.

#### Does this pull request change what data or activity we track or use?


NO

#### Testing instructions:






Regression Testing
1) Verify that new site registration have home_url and site_url as expected on the WP.com Cache (E2E flows cover)
2) Verify that Identity Crisis flows are fully functional 
 -- Duplicate your JP connected site to a new site (update options to new url except leave jetpack_private_options untouched)
 -- Perform an edit to a post (triggers sync)
 -- Visit the Jetpack Dashboard
 -- Safe Mode will be enabled and IDC options presented.
 -- Verify that choosing this is a new site, doesn't modify the home_url or site_url of the existing cache site and instead the new site has its own cache site created and jetpack_private_options updated. 
 -- Verify that choosing this is the site has stats migrated and home_url and site_url of the existing cache site is updated.
3) Modify a site's site_url and home_url. Ensure these are synced to the WP.com Cache (unit tests cover)

Note: Quickest way to trigger IDC flows is to copy jetpack_private_options from Site A to Site B then trigger a content change on Site B (edit post, etc)
